kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/main/java/org/kie/workbench/common/services/refactoring/backend/server/indexing/PackageDescrIndexVisitor.java

```
@@ -188,101 +233,117 @@ private void visit( final Object descr ) {
-            factType = addField( fieldName,
-                                 fullyQualifiedClassName );
-            if ( factType != null ) {
-                fullyQualifiedClassName = getFullyQualifiedClassName( packageDescr,
-                                                                      factType );
-            }
+
+    protected void visit( final AttributeDescr descr ) {
+        String attr = descr.getName();
+        switch(attr) {
+            case "salience":
+            case "dialect":
+                return;
         }
-        return addField( expression,
-                         fullyQualifiedClassName );
-    }
 
-    private String addField( final String fieldName,
-                             final String fullyQualifiedClassName ) {
-        final ModelField[] mfs = dmo.getProjectModelFields().get( fullyQualifiedClassName );
-        if ( mfs != null ) {
-            for ( ModelField mf : mfs ) {
-                if ( mf.getName().equals( fieldName ) ) {
-                    builder.addGenerator( new TypeField( new ValueFieldIndexTerm( fieldName ),
-                                                         new ValueTypeIndexTerm( mf.getClassName() ),
-                                                         new ValueTypeIndexTerm( fullyQualifiedClassName ) ) );
-                    return mf.getClassName();
-                }
-            }
+        PartType type = PartType.getPartTypeFromAttribueDescrName(descr.getName());
+        switch(type) {
+            case AGENDA_GROUP:
+            case ACTIVATION_GROUP:
+            case RULEFLOW_GROUP:
+            case ENTRY_POINT:
+                SharedPart sharedRef = new SharedPart(descr.getValue(), type);
+                builder.addGenerator( sharedRef );
+            break;
+            default:
+                throw new UnsupportedOperationException("Unsupported attribute encountered: " + descr.getName() );
```

Really!?!? There are lots more attributes than those above; ```salience``` for example.. So I ask, swallow or throw the exception?